### DISCUSSION TITLE
How to delete full json object

### DISCUSSION THREAD

**Discussion by ZeroXKiritsu** [nlohmann/json#3714] [Q&A] [answered]
I want to delete the json object completely so that there is no empty object left, that's how I delete json, but after deleting, an empty object remains, its possibly? group - its a json 
```cpp
for (auto &group : settings)
{
        if (group[GroupKey] == groupKey)
            group.clear();
}
```
before deleting:
```json
{
"groupKey": "Example",
"groupName": "Example",
"groupElements": []
}
```
after deleting:
```json
{}
```
I want see this: 
```
```

**Comment by falbrechtskirchinger** [comment]
Please provide a complete, compilable example, as well as the start and end JSON values. Your description is somewhat ambiguous.

**Reply by ZeroXKiritsu** [reply]
i update question

**Comment by falbrechtskirchinger** [comment, marked as answer]
Assuming `settings` is an array, you can write:
```cpp
for(auto it = settings.begin(); it != settings.end(); ) {
    if(it->at(GroupKey) == groupKey)
        it = settings.erase(it);
    else
        ++it;
}
```

https://godbolt.org/z/bMahhPPxn

**Reply by ZeroXKiritsu** [reply]
Thank you
